### PR TITLE
use service title in image name

### PR DESCRIPTION
**Pull request by hugoesthere**
Use `author/service-title` naming convention for docker images

@kevgo @trushton @martinjaime

### CODE REVIEW

File: exo-deploy/src/docker/docker-hub.ls [hunk 1]
```diff
@@ -40,7 +40,8 @@ class DockerHub
         service-config = require path.join(process.cwd!, config.location, 'service.yml')
         names.push do
           author: service-config.author
-          name: path.basename config.location
+          name: service-config.title |> (.replace /\s/g, '-')
```
Comment (kevgo) [inline exo-deploy/src/docker/docker-hub.ls]: Use https://www.npmjs.com/package/kebab-case to translate the string properly into kebab-case.

File: exo-deploy/src/docker/docker-hub.ls [hunk 2]
```diff
@@ -40,7 +40,8 @@ class DockerHub
...
+          #TODO: get image name if location is docker on dockerhub
```
[the remark this answers is not in the document]
Comment (kevgo) [reply exo-deploy/src/docker/docker-hub.ls:46]: Tododrop is right, this should be a ticket on Github!

File: exo-deploy/src/terraform/aws-terraform-file-builder.ls [hunk 3]
```diff
@@ -71,7 +71,8 @@ class AwsTerraformFileBuilder
   _generate-services: (type) ->
     for service-name, service-data of @app-config.services["#{type}"]
       service-config = require path.join('/var/app', service-data.location, 'service.yml')
-      @_build-service-container-definition service-name, service-config
+      image-name = @_get-image-name service-data
```
Comment (kevgo) [inline exo-deploy/src/terraform/aws-terraform-file-builder.ls]: I think you can inline this variable

**Review by kevgo — APPROVED** [review]

File: exo-deploy/src/docker/docker-hub.ls (hunk 1 above)
Comment (hugoesthere) [reply exo-deploy/src/docker/docker-hub.ls]: This is converting whitespace separated titles to kebab-case. That package will convert camel-case to kebab-case only. Should we standardize what format service titles can be in?

Comment (kevgo) [reply exo-deploy/src/docker/docker-hub.ls]: Okay, then use https://www.npmjs.com/package/dashify. I'd rather leave the format of service titles open.

**Pull request closed without merging**